### DISCUSSION TITLE
Working with Worklogs

### DISCUSSION THREAD

**Discussion by osokin** [andygrunwald/go-jira#367] [Q&A] [answered]
Hi,

thanks very much for the application.

I've found a way how to get issues with a filter and I'm looking for a way to get all worklogs for those issues now.
The ` IssueFields` structure has the `*Worklog` pointer, and the `Worklog` structure has the `Worklogs` array of the `WorklogRecord` structures.

Could you please provide a code sample or a snippet to demonstrate access to the WorklogRecord structure's fiels.

Thank you.

**Comment by manuelbcd** [comment, marked as answer]
Hello @osokin.   I built a simple example to gather work-logs from jira issues and export them to excel file: https://github.com/manuelbcd/jira-logtime-exporter

let me invite you to have a look to the source code.  Here you have an extract

```
       tp := jira.BasicAuthTransport{
		Username: cfg.Jira.Login,
		Password: cfg.Jira.Token,
	}

	jiraClient, _ := jira.NewClient(tp.Client(), cfg.Jira.Host)

	// Removed dateStart property since Jira API has a known bug. TODO: reactivate once the bug is fixed.
	// dateStart := int64(time.Now().Unix())
	// var op * jira.GetWorklogsQueryOptions = &jira.GetWorklogsQueryOptions{Expand: "properties", StartedAfter: dateStart}

	var op * jira.GetWorklogsQueryOptions = &jira.GetWorklogsQueryOptions{Expand: "properties"}
	issue, _, err := jiraClient.Issue.GetWorklogs(issueid, jira.WithQueryOptions(op))

	if err != nil {
		fmt.Printf("issuePtr: %s \n",err.Error())
		os.Exit(1)
	}

	saveToExcelFile(issue, dir)
```

Please note: Query Options are not working for WorkLogs requests due to an Atlassian known bug

Let me know if this is what you was looking for or you need help with other topic

Best.

**Reply by osokin** [reply]
Hi @manuelbcd , thanks for the quick reply.
I'm looking for a way to automate an accounting process follow to a data from working logs records, but without an external intervention by third-party tools.

So, the issues have been received with a code that very closer to the `GetAllIssues()` function, you have in examples.
Since we may have one or more records in `*Worklogs`, I'm interested to see an example that demonstrates how to work with those.

Thank you.

**Reply by manuelbcd** [reply]
I think what you need is to iterate over the list of issues in the result and request their WorkLogs one by one.

To request issues and iterate over them you can check this example
```
	for _, i := range issues {
		fmt.Printf("%s (%s/%s): %+v\n", i.Key, i.Fields.Type.Name, i.Fields.Priority.Name, i.Fields.Summary)
	}
```
Check full example here: https://github.com/andygrunwald/go-jira/blob/master/examples/jql/main.go

And the code to request worklogs from each issue ID is what I pasted above, basically
```
var op * jira.GetWorklogsQueryOptions = &jira.GetWorklogsQueryOptions{Expand: "properties"}
issue, _, err := jiraClient.Issue.GetWorklogs(issueid, jira.WithQueryOptions(op))`
```

Does this resolve your question?

**Reply by osokin** [reply]
Thank you very much!

**Comment by osokin** [comment]
That works, thanks!
